service/storage_proxy.cc

```diff
@@ -3094,6 +3094,14 @@ class digest_read_resolver : public abstract_read_resolver {
         auto& first = *_digest_results.begin();
         return std::find_if(_digest_results.begin() + 1, _digest_results.end(), [&first] (digest_and_last_pos digest) { return digest.digest != first.digest; }) == _digest_results.end();
     }
+    rpc::optional<full_position> min_position() {
+        return std::min_element(_digest_results.begin(), _digest_results.end(), [this] (const digest_and_last_pos& a, const digest_and_last_pos& b) {
+            if (!a.last_pos || !b.last_pos) {
+                return (bool(a.last_pos) - bool(b.last_pos)) < 0;
```

Comment:
gcc 12 will whine, so int(bool(a.last_pos)).

Reply:
also, the subtraction is unneeded, you can compare the bools directly.

What does it mean if one result has a last position and the other doesn't?

Reply:
This is only possible in mixed cluster: some replicas don't set last position, while some do. We want to sort the un-engaged position as the smallest one, so it is the position we set on the final result (meaning, there is no replica-provided last-position).

Reply:
changed to compare bools